src/components/Sections/SimpleMode/list.js

```diff
@@ -2,44 +2,54 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { withNamespaces } from 'react-i18next';
 import { Image, List } from 'semantic-ui-react';
+import { useSelector } from 'react-redux';
 
+import WipErr from '../../shared/WipErr/WipErr';
+import { FrownSplash } from '../../shared/Splash/Splash';
+import { selectors } from '../../../redux/modules/simpleMode';
+import { selectors as mdb } from '../../../redux/modules/mdb';
+import { isEmpty } from '../../../helpers/utils';
 import { SectionLogo } from '../../../helpers/images';
-import * as shapes from '../../shapes';
 
-const SimpleModeList = ({ items = {}, language, t, renderUnit }) => (
-  <div>
-    {
-      items.lessons.length
-        ? (
-          <div>
-            <h2>
-              <Image className="simple-mode-type-icon">
-                <SectionLogo name='lessons' />
-              </Image>
-              {t('simple-mode.today-lessons')}
-            </h2>
-            <List size="large">
-              {items.lessons.map(x => renderUnit(x, language, t))}
-            </List>
-          </div>
-        )
-        : null
-    }
+const SimpleModeList = ({ language, t, renderUnit }) => {
+  const wip = useSelector(state => selectors.getWip(state.simpleMode));
+  const err = useSelector(state => selectors.getError(state.simpleMode));
+  const reduxItems = useSelector(state => selectors.getItems(state.simpleMode));
+  const lessons = useSelector(state => reduxItems.lessons.map(x => mdb.getDenormCollectionWUnits(state.mdb, x)).filter(x => !isEmpty(x)));
+  const others = useSelector(state => reduxItems.others.map(x => mdb.getDenormContentUnit(state.mdb, x)).filter(x => !isEmpty(x)));
 
-    {
-      items.others.length
-        ? (
-          <List size="large">
-            {renderUnit(items.others, language, t)}
-          </List>
-        )
-        : null
-    }
-  </div>
-);
+  return (
+    WipErr({ wip, err, t }) ||
```

Comment:
Please make this code more simple.
const wipErr = WipErr({ wip, err, t });
  if (wipErr) {
    return wipErr;
  }

if (lessons.length === 0 && others.length === 0) {
  return <FrownSplash text={t('simple-mode.no-files-found-for-date')} />;
}

And then this code without the above conditions

Reply:
what you suggest is more complicated...